jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -215,14 +221,42 @@ class ProfileCredentialProviderFactory(private val ssoCache: SsoCache = diskCach
     private fun createSsoProvider(profile: Profile): AwsCredentialsProvider = ProfileSsoProvider(profile)
 
     private fun createAssumeRoleProvider(profile: Profile, region: AwsRegion): AwsCredentialsProvider {
-        val sourceProfileName = profile.requiredProperty(ProfileProperty.SOURCE_PROFILE)
-        val sourceProfile = profileHolder.getProfile(sourceProfileName)
-            ?: throw IllegalStateException("Profile $sourceProfileName looks to have been removed")
-        val parentCredentialProvider = createAwsCredentialProvider(sourceProfile, region)
+        val sourceProfileName = profile.property(ProfileProperty.SOURCE_PROFILE)
+        val credentialSource = profile.property(ProfileProperty.CREDENTIAL_SOURCE)
+
+        val parentCredentialProvider = when {
+            sourceProfileName.isPresent -> {
+                val sourceProfile = profileHolder.getProfile(sourceProfileName.get())
+                    ?: throw IllegalStateException("Profile $sourceProfileName looks to have been removed")
+                createAwsCredentialProvider(sourceProfile, region)
+            }
+            credentialSource.isPresent -> {
+                // Can we parse the credential_source
+                credentialSourceCredentialProvider(CredentialSourceType.parse(credentialSource.get()), profile)
+            }
+            else -> {
+                throw IllegalArgumentException(message("credentials.profile.assume_role.missing_source", profile.name()))
+            }
+        }
 
         return ProfileAssumeRoleProvider(parentCredentialProvider, region, profile)
     }
 
+    private fun credentialSourceCredentialProvider(credentialSource: CredentialSourceType, profile: Profile): AwsCredentialsProvider =
+        when (credentialSource) {
```

Comment:
is there a preferred order documented anywhere or is this arbitrary?

Reply:
it is an enum, so order does not matter